sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -458,12 +459,18 @@ void gnrc_ndp_rtr_sol_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
                 /* or unicast, if source is known */
                 /* XXX: can't just use GNRC_NETAPI_MSG_TYPE_SND, since the next retransmission
                  * must also be set. */
-                gnrc_ipv6_nc_t *nc_entry = gnrc_ipv6_nc_get(iface, &ipv6->src);
+                nc_entry = gnrc_ipv6_nc_get(iface, &ipv6->src);
                 xtimer_set_msg(&nc_entry->rtr_adv_timer, delay, &nc_entry->rtr_adv_msg,
                                gnrc_ipv6_pid);
             }
 #endif
         }
+        nc_entry = gnrc_ipv6_nc_get(iface, &ipv6->src);
+        if (nc_entry != NULL) {
```

Comment:
Sorry, but do you have a better explanation than "trust me!"? ;-)
In line 462 `gnrc_ipv6_nc_get()` gets called to get the neighbor cache entry for the source address. I fail to see why this should always succeed there. What if someone removes the neighbor cache entry manually (using the shell) just in this moment?